enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -615,21 +663,43 @@ func (p *PebbleCache) scanForBrokenFiles(quitChan chan struct{}) error {
 			log.Errorf("Error unmarshaling metadata when scanning for broken files: %s", err)
 			continue
 		}
-		blobDir = p.blobDir(fileMetadata.GetFileRecord().GetIsolation().GetPartitionId())
 
-		_, err := p.fileStorer.NewReader(p.env.GetServerContext(), blobDir, fileMetadata.GetStorageMetadata(), 0, 0)
-		if err != nil {
-			if p.handleMetadataMismatch(p.env.GetServerContext(), err, fileMetadataKey, fileMetadata) {
-				mismatchCount += 1
+		removedEntry := false
+		if opts.deleteEntriesWithMissingFiles {
+			blobDir = p.blobDir(fileMetadata.GetFileRecord().GetIsolation().GetPartitionId())
+			_, err := p.fileStorer.NewReader(p.env.GetServerContext(), blobDir, fileMetadata.GetStorageMetadata(), 0, 0)
+			if err != nil {
+				if p.handleMetadataMismatch(p.env.GetServerContext(), err, fileMetadataKey, fileMetadata) {
+					missingFiles += 1
+					removedEntry = true
+				}
 			}
 		}
 
-		if fileMetadata.GetFileRecord().GetCompressor() == repb.Compressor_IDENTITY {
+		if opts.deleteACEntriesOlderThan != 0 {
+			age := time.Since(time.UnixMicro(fileMetadata.GetLastAccessUsec()))
+			if age > opts.deleteACEntriesOlderThan {
+				if err := db.Delete(fileMetadataKey, pebble.NoSync); err != nil {
```

Comment:
i'm prolly just missing where it happens -- but how does this code check that it's deleting an AC entry?

Reply:
Err oops.